HealthcareSystem/Backend/Model/Users/City.cs

```diff
@@ -5,13 +5,17 @@
  ***********************************************************************/
 
 using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
 namespace Model.Users
 {
     public class City
     {
-        public string Name { get; set; }
+        [Key]
         public int ZipCode { get; set; }
+        public string Name { get; set; }
+        public virtual ICollection<User> Users { get; set; }
```

Comment:
Have you tried using the [manual configuration for one to many relationships described here](https://docs.microsoft.com/en-us/ef/core/modeling/relationships?tabs=fluent-api%2Cfluent-api-simple-key%2Csimple-key#manual-configuration) instead of adding the collection properties? I haven't tried but apparently it should be possible to use this and just leave the 'many' side empty, if it works it might be easier than doing it like this for the entire model. 
I believe it should go something like:
modelBuilder.Entity<User>()
            .HasOne(p => p.City)
            .WithMany();
I'm not sure if user being abstract will make issues here, but it should work for Feedback and Patient at least?